src/cmd_line/commandLineHistory.ts

```diff
@@ -5,7 +5,7 @@ import { Logger } from '../util/logger';
 
 export class CommandLineHistory {
   private _history: string[] = [];
-  private _is_loaded: boolean = false;
+  private _is_loading: boolean = false;
```

Comment:
Can you turn this into a promise chain/queue instead? Basically, each time save or load is called, chain it onto the last promise with `finally` and set the member to the new promise.
- You won't have to fail in save because it will never run at the same time as a load
- You won't have to check for new history in load because when it runs it will be the latest operation
- Saves will always write in order, which is more important if this ever gets changed to just appending to the history file or for slow disks.